src/types/string.rs

```diff
@@ -144,6 +144,33 @@ impl PyString {
         unsafe { py.from_owned_ptr(ffi::PyUnicode_FromStringAndSize(ptr, len)) }
     }
 
+    /// Intern the given string
+    ///
+    /// This will return a reference to the same Python string object if called repeatedly with the same string.
+    ///
+    /// Note that a temporary allocation is required if the given string is not already null-terminated.
```

Comment:
This will always allocate though? Because a PyString is always created whether or not it is already interned.

I understand this isn't more performant than just using `PyString::new`, only more memory efficient? That should be mentioned IMO.

Reply:
The allocation of the CString is meant here.

Reply:
As written above, that the current implementation provided by CPython unconditionally allocates is a bug IMHO.